indra/databases/cbio_client.py

```diff
@@ -88,12 +89,92 @@ def get_mutations(study_id, gene_list, mutation_type=None,
     if case_id:
         df = df[df['case_id'] == case_id]
     res = _filter_data_frame(df, ['gene_symbol', 'amino_acid_change'],
-                                   'mutation_type', mutation_type)
+                             'mutation_type', mutation_type)
     mutations = {'gene_symbol': list(res['gene_symbol'].values()),
                  'amino_acid_change': list(res['amino_acid_change'].values())}
     return mutations
 
 
+def get_case_lists(study_id):
+    """Return a list of the case set ids for a particular study.
+
+    TAKE NOTE the "case_list_id" are the same thing as "case_set_id"
+    Within the data, this string is referred to as a "case_list_id".
+    Within API calls it is referred to as a 'case_set_id'.
+    The documentation does not make this explicitly clear.
+
+    Parameters
+    ----------
+    study_id : str
+        The ID of the cBio study.
+        Example: 'cellline_ccle_broad' or 'paad_icgc'
+
+    Returns
+    -------
+    case_set_ids : dict[dict[int]]
+        A dict keyed to cases containing a dict keyed to genes
+        containing int
+    """
+    data = {'cmd': 'getCaseLists',
+            'cancer_study_id': study_id}
+    df = send_request(**data)
+    case_set_ids = df['case_list_id'].tolist()
+    return case_set_ids
+
+
+def get_profile_data(study_id, gene_list, profile_filter, case_set_id=None):
+    """Return dict of cases and genes and their respective values.
+
+    Parameters
+    ----------
+    study_id : str
+        The ID of the cBio study.
+        Example: 'cellline_ccle_broad' or 'paad_icgc'
+    gene_list : list[str]
+        A list of genes with their HGNC symbols.
+        Example: ['BRAF', 'KRAS']
+    profile_filter : str
+        A string used to filter the profiles to return. Will be one of:
+        - MUTATION
+        - MUTATION_EXTENDED
+        - COPY_NUMBER_ALTERATION
+        - MRNA_EXPRESSION
+        - METHYLATION
+    case_set_id : Optional[str]
```

Comment:
I'd rename this argument to case_set_filter because it acts as a filter. Then lines 156-157 would be
```
if case_set_filter:
    case_set_id = [x for x in case_set_ids if case_set_filter in x][0]
```

Reply:
Renamed parameter in code and tests with commit c22f95b